client/ayon_core/tools/publisher/control.py

```diff
@@ -150,10 +150,15 @@ def _create_plugin_data_item(self, plugin):
         label = None
         if hasattr(plugin, "label"):
             label = plugin.label
-
+        result = inspect.getmro(plugin)
+        plugin_type = result[1].__name__
```

Comment:
What is `plugin_type`?

Reply:
I guess the idea was for it to be `InstancePlugin` or `ContextPlugin` according to this screenshot: https://github.com/ynput/ayon-core/pull/142#issuecomment-1975322813

However, it always being the second entry in the MRO order I think isn't what we should rely on.

I feel like we're much better of here doing something like:

```python
if isinstance(plugin, pyblish.api.InstancePlugin):
    plugin_type = "InstancePlugin"
elif isinstance(plugin, pyblish.api.ContextPlugin):
    plugin_type = "ContextPlugin"
else:
    plugin_type = "unknown"
```

Or `issubclass` if the `plugin` variable is the class and not an instance.

Reply:
In that case we have to face fact, that in pyblish you can't rely on class inheritance.

```python
if plugin.__instanceEnabled__:
    plugin_type = "Instance plugin"
else:
    plugin_type = "Context plugin"
```

Reply:
Yes, thanks! :)

Reply:
Is this how I should distinguish between plugin types
```
if plugin.__instanceEnabled__:
    plugin_type = "Instance plugin"
else:
    plugin_type = "Context plugin"
```
Does this really work reliably?

Reply:
thanks this seems to work reliably
```
if plugin.__instanceEnabled__:
    plugin_type = "Instance plugin"
else:
    plugin_type = "Context plugin"
```